File: modules/security/exercises/03_secret_scan/solution/python/main.py
import re

def main() -> None:
    code = 'api_key = "sk-test123456789"'
    if re.search(r"sk-[A-Za-z0-9_-]+", code):
        print("SECRET_SCAN_OK")

if __name__ == "__main__":
    main()

File: modules/security/exercises/03_secret_scan/solution/cpp/main.cpp
#include <iostream>
#include <regex>
#include <string>

int main() {
    std::string code = R"(api_key = "sk-test123456789")";
    if (std::regex_search(code, std::regex(R"(sk-[A-Za-z0-9_-]+)"))) {
        std::cout << "SECRET_SCAN_OK\n";
    }
    return 0;
}
